### PR TITLE
Move the deployment check into common code and add tests.

### PR DESCRIPTION
**Pull request by markusthoemmes**
## Proposed Changes

As per title, this is the first function to be pulled out into common code. It makes the DeploymentCheck available from either Serving and Eventing and adds unit tests to the formerly untested code.

**Release Note**

```release-note
NONE
```

/assign @jcrossley3 @aliok @houshengbo

### CODE REVIEW

File: pkg/reconciler/knativeeventing/knativeeventing.go [hunk 1]
```diff
@@ -118,7 +115,9 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, ke *eventingv1alpha1.Kna
 	stages := []func(context.Context, *mf.Manifest, *eventingv1alpha1.KnativeEventing) error{
 		r.ensureFinalizerRemoval,
 		r.install,
-		r.checkDeployments,
+		func(ctx context.Context, manifest *mf.Manifest, ke *eventingv1alpha1.KnativeEventing) error {
+			return common.CheckDeployments(r.kubeClientSet, manifest, &ke.Status)
+		},
```
Comment (jcrossley3) [inline pkg/reconciler/knativeeventing/knativeeventing.go:120]: Personally, I'd rather see a one-line member function than a three-line anonymous one. So I'd vote for keeping `r.checkDeployments` and have it do the `common.CheckDeployments` call.

File: pkg/reconciler/knativeserving/knativeserving.go [hunk 2]
```diff
@@ -118,7 +115,9 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, ks *servingv1alpha1.Knat
 	stages := []func(context.Context, *mf.Manifest, *servingv1alpha1.KnativeServing) error{
 		r.ensureFinalizerRemoval,
 		r.install,
-		r.checkDeployments,
+		func(ctx context.Context, manifest *mf.Manifest, ks *servingv1alpha1.KnativeServing) error {
+			return common.CheckDeployments(r.kubeClientSet, manifest, &ks.Status)
+		},
```
Comment (jcrossley3) [inline pkg/reconciler/knativeserving/knativeserving.go:120]: ibid

File: pkg/reconciler/knativeeventing/knativeeventing.go (hunk 1 above)
Comment (Cynocracy) [reply pkg/reconciler/knativeeventing/knativeeventing.go]: +1

**Comment by Cynocracy** [comment]
/retest

Overall this looks good :) +1 to Jim's comment. Will lgtm once the tests pass.

**Pull request merged**